action.yml

```
@@ -36,43 +36,107 @@ outputs:
+            echo "IS_V2=$(echo "$configuration_response" | jq -r .v2_enabled)" >> $GITHUB_ENV
+            echo "IS_V1=$(echo "$configuration_response" | jq -r .v1_enabled)" >> $GITHUB_ENV
+          fi
         else
-          echo "IS_V2=false" >> $GITHUB_ENV
+          echo "Single deployment..."
+          if [[ "${TOKEN}" =~ ^[A-Za-z0-9+/]+[=]{0,2}$ ]]; then
+            echo "Token is in V2 format..."
+            echo "IS_V2=true" >> $GITHUB_ENV
+            echo "PRIORITY=v2" >> $GITHUB_ENV
+            echo "v2_token=${TOKEN}" >> $GITHUB_OUTPUT
+          else
+            echo "Token is in V1 format..."
+            echo "IS_V1=true" >> $GITHUB_ENV
+            echo "PRIORITY=v1" >> $GITHUB_ENV
+            echo "v1_token=${TOKEN}" >> $GITHUB_OUTPUT
+          fi
         fi
 
-    - name: Build and Publish to Oxygen V2
-      shell: 'bash'
-      id: 'oxygen-cli-action'
-      if: env.IS_V2 == 'true'
-      run: |
-        yarn global add @shopify/oxygen-cli@latest
-        URL_OUTPUT=$(oxygen-cli oxygen:deploy \
-          --path=${{ inputs.path }} \
-          --assetsFolder=${{ inputs.oxygen_client_dir }} \
-          --workerFolder=${{ inputs.oxygen_worker_dir }})
-        echo "url=$URL_OUTPUT" >> $GITHUB_OUTPUT
-      env:
-        OXYGEN_DEPLOYMENT_TOKEN: ${{ inputs.oxygen_deployment_token }}
-
-    - name: Build and Publish to Oxygen V1
+    - name: Build and Publish to Oxygen
       shell: 'bash'
       id: 'oxygenctl-action'
-      if: env.IS_V2 == 'false'
+      if: always()
       env:
         OXYGEN_ASSETS_DIR: ${{ inputs.oxygen_client_dir }}
         OXYGEN_BUILD_COMMAND: ${{ inputs.build_command }}
         OXYGEN_COMMIT_MESSAGE: ${{ inputs.commit_message || 'No commit message' }}
-        OXYGEN_DEPLOYMENT_TOKEN: ${{ inputs.oxygen_deployment_token }}
         OXYGEN_WORKFLOW_ID: "${{ github.run_id }}_${{ github.run_attempt }}"
         OXYGEN_WORKER_DIR: ${{ inputs.oxygen_worker_dir }}
       run: |
-        [[ -n "${{ inputs.path }}" ]] && cd "${{ inputs.path }}";
-        [[ -z "${{ inputs.commit_timestamp }}" ]] && export OXYGEN_COMMIT_TIMESTAMP=$(date -u +"%Y-%m-%dT%H:%M:%SZ") || export OXYGEN_COMMIT_TIMESTAMP=${{ inputs.commit_timestamp }};
-        echo "url=$($GITHUB_ACTION_PATH/oxygenctl deploy --health-check='${{ inputs.oxygen_health_check }}')" >> $GITHUB_OUTPUT
+        [[ -n "${{ inputs.path }}" ]] && cd "${{ inputs.path }}"
+
+        if [ "${{ env.IS_V2 }}" == "true" ]; then
+          echo "Deploying..."
+          mkdir oxygen_v2
+          yarn global add @shopify/oxygen-cli@latest
+          rsync -a ./* ./oxygen_v2/ --exclude 'oxygen_v2' --exclude 'node_modules'
+
+          oxygen_v2_command="oxygen-cli oxygen:deploy \
+                --path=${{ inputs.path }} \
+                --assetsFolder=${{ inputs.oxygen_client_dir }} \
+                --workerFolder=${{ inputs.oxygen_worker_dir }} \
+                --token=${{ steps.check_version.outputs.v2_token }}"
```

```suggestion
          build_command_filtered=$(echo "${{ inputs.build_command }}" | sed 's/HYDROGEN_ASSET_BASE_URL=$OXYGEN_ASSET_BASE_URL //g')
          oxygen_v2_command="oxygen-cli oxygen:deploy \
                --path=${{ inputs.path }} \
                --assetsFolder=${{ inputs.oxygen_client_dir }} \
                --workerFolder=${{ inputs.oxygen_worker_dir }} \
                --token=${{ steps.check_version.outputs.v2_token }}" \
                --buildCommand='$build_command_filtered'"
```
untested, but would something like this work? It would save us from needing this in the CLI itself (https://github.com/Shopify/oxygen-cli/pull/111).

---

👍🏻  Will give it a test

---

Added 🥳